siad/daemonapi.go

```diff
@@ -22,26 +27,28 @@ const VERSION = "0.2.0"
 // manifest.
 var updateURL = "http://23.239.14.98/releases/" + runtime.GOOS + "_" + runtime.GOARCH
 
-// returns true if version is "greater than" VERSION.
+// newerVersion returns true if version is "greater than" VERSION.
 func newerVersion(version string) bool {
-	// super naive; assumes same number of .s
-	// TODO: make this more robust... if it's worth the effort.
-	nums := strings.Split(version, ".")
 	NUMS := strings.Split(VERSION, ".")
-	for i := range nums {
-		// inputs are trusted, so no need to check the error
-		ni, _ := strconv.Atoi(nums[i])
+	nums := strings.Split(version, ".")
+	for i := range NUMS {
 		Ni, _ := strconv.Atoi(NUMS[i])
-		if ni != Ni {
-			return ni > Ni
+		ni, _ := strconv.Atoi(nums[i])
+		if Ni != ni {
+			return Ni < ni
+		}
+		if len(ni)-1 == i {
+			return false
 		}
 	}
-	// versions are equal
+	if len(NUMS) < len(nums) {
+		return true
+	}
```

Comment:
although I think if you do the truncation I mentioned above, you don't need this check anymore